### DISCUSSION TITLE
v0.2.0

### DISCUSSION THREAD

**Discussion by erogol** [coqui-ai/TTS#720] [Versions & Releases]
# 🐸 v0.2.0

### 🐞Bug Fixes 

- Fix phoneme pre-compute issue.
- Fix multi-speaker setup in Tacotron models.
- Fix small issues in the Trainer regarding multi-optimizer training.

### 💾 Code updates

- W&B integration for model logging and experiment tracking, (👑 @AyushExel)
      Code uses the Tensorboard by default. For W&B, you need to set `log_dashboard` option in the config and define `project_name` and `wandb_entity`. 
- Use ffsspec for model saving/loading (👑 @agrinh)
- Allow models to define their own symbol list with in-class `make_symbols()`
- Allow choosing after epoch or after step LR scheduler update with `scheduler_after_epoch`.
- Make converting spectrogram from amplitude to DB optional with `do_amp_to_db_linear` and `do_amp_to_db_linear` options.

### 🗒️  Docs updates

- Add GlowTTS and VITS docs.

### 🤖 Model implementations

- VITS implementation with pre-trained models (https://arxiv.org/abs/2106.06103)

### 🚀 Model releases

- vocoder_models--ja--kokoro--hifigan_v1 (👑 @kaiidams)

    HiFiGAN model trained on Kokoro dataset to complement the existing Japanese model. 

    ### Try it out:
    ```bash
    tts --model_name tts_models/ja/kokoro/tacotron2-DDC --text "こんにちは、今日はいい天気ですか？"
    ``` 
 
- tts_models--en--ljspeech--tacotronDDC_ph

    TacotronDDC with phonemes trained on LJSpeech. It is to fix the pronunciation errors caused by the raw text
    in the released TacotronDDC model.

    ### Try it out:
    ```bash
    tts --model_name tts_models/en/ljspeech/tacotronDDC_ph --text "hello, how are you today?"
    ``` 

- tts_models--en--ljspeech--vits

    VITS model trained on LJSpeech.

    ### Try it out:
    ```bash
    tts --model_name tts_models/en/ljspeech/vits --text "hello, how are you today?"
    ```

- tts_models--en--vctk--vits

    VITS model trained on VCTK with multi-speaker support.

    ### Try it out:
    ```bash
    tts-server --model_name tts_models/en/vctk/vits     
    ```

- vocoder_models--en--ljspeech--univnet

    UnivNet model trained on LJSpeech to complement the TacotronDDC model above.

    ### Try it out:
    ```bash
    tts --model_name tts_models/en/ljspeech/tacotronDDC_ph --text "hello, how are you today?"
    ```

**Comment by godspirit00** [comment]
Thanks for the great work! 
I have a question. I trained a TTS model in a previous version, and I noticed some pronunciation errors in the generated speech. Now that version 0.2.0 fixed it, do I have to retrain the model from start? Is it ok to continue training from the last checkpoint to fix the problem? 
Thanks!

**Reply by erogol** [reply]
probably enough to finetune

**Comment by Sadam1195** [comment]
Just a suggestion, it would be nice to have some information on models released in every version. Like what dataset was used to train, number of steps, GPU resources used (time taken : a bit irrelevant but gives some idea about what to expect). Sometimes it is self explanatory with the model name but nonetheless, it would be good to have official statement on it.

**Reply by BillyBobQuebec** [reply]
I second this! Nvidia model catalog for example has very helpful details about what gpu(s) and batch size was used to train a model and what loss the model was at within different points of training, as well as inference speed on gpu vs cpu...

**Reply by erogol** [reply]
The info I wrote is from the paper. Depending on your local settings can things change. But I've not run apple to apple comparison personally. 

Model cards are hard to maintain right now for a library in v0.2.0. But we plan to set it up when things are more stable.
